cse/WpOldApiAdapter.py

```diff
@@ -0,0 +1,158 @@
+import requests
+import os
+from string import Template
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+     from urlparse import urlparse
+
+from cse.pipeline import Handler
+from cse.util import Util
+
+class WpOldApiAdapter(Handler):
+
+    API_MUX_ENDPOINT = "https://comments-api.ext.nile.works/v2/mux"
+    API_SEARCH_ENDPOINT = "https://comments-api.ext.nile.works/v1/search"
+
+    __handlerContext = None
+
+    def __init__(self):
+        super()
+
+    def loadComments(self, url):
+        # if self.__handlerContext is None:
+        #     raise Exception("WpApiAdapter must be used within a WpApiAdapterHandler to use pipelining functionality!")
+        
+        data = self.__loadInitialRootComments(url)
+        
+        comments = self.__processComments(data, url)
+        #print(comments)
+
+        while data['hasMoreChildren'] == "true":
+            data = self.__loadMoreRootComments(url, data['nextPageAfter'])
+            comments.update(self.__processComments(data, url))
+
+        commentsList = self.__buildDataSkeleton(url)
+        commentsList["comments"] = comments
+        print(len(comments))
+
+    def __buildDataSkeleton(self, url, assetId=None):
+        return {
+            "article_url" : url,
+            "article_id" : assetId,
+            "comments" : []
+        }
+
+    def __processComments(self, data, url):
+        commentList = {}
+
+        for entry in data["entries"]:
+            cid = self.__extractCid(entry["object"]["id"])
+            comment_author = entry["actor"]["title"]
+            comment_text = entry["object"]["content"]
+            timestamp = entry["object"]["published"]
+
+            for target in entry["targets"]:
+                try:
+                   parent_comment_id = self.__extractCid(target["conversationID"])
+                   break
+                except KeyError:
+                    pass
+            if parent_comment_id == cid:
+                parent_comment_id = None
+
+            try:
+                votes = entry["object"]["accumulators"]["likesCount"]
+            except KeyError:
+                votes = 0
+
+            commentList[cid] = {
+                "comment_author": comment_author,
+                "comment_text" : comment_text,
+                "timestamp" : timestamp,
+                "parent_comment_id" : parent_comment_id,
+                "votes" : votes
+            }
+            try:
+                directReplies = self.__loadReplies(entry["object"]["id"], entry["object"]["accumulators"]["repliesCount"], entry["pageAfter"])
```

Comment:
When I studied the result data structure of the old API, the replies of comments were directly in the root comment list (right below the parent comment), so no additional request was needed. Can you check this again? Maybe this is the cause of the additional comments in our list compared to the displayed total number of comments?

Reply:
Actually just 3 replies per layer are displayed by default. I thought it's easier to get no replies in the beginning and to request all of them if there are any instead of dealing with checking which comments were already processed.